syntax = "proto2";

package auth;

message CreateAccountWeb {
    optional int64 requestId = 1;
    optional string email = 2;
    optional string plaintextPassword = 3;
}

message CreateAccountWebSuccess {
    optional int64 requestId = 1;
    optional int64 userId = 2;
}

message CreateAccountWebFailure {
    optional int64 requestId = 1;
    enum FailureReason {
        ACCOUNT_ALREADY_EXISTS = 0;
        INVALID_PASSWORD = 1;
        INTERNAL_SERVER_ERROR = 2;
    }

    optional FailureReason reason = 2;
}

message AuthenticateWeb {
    optional int64 requestId = 1;
    optional string email = 2;
    optional string plaintextPassword = 3;
}

message AuthenticateWebSuccess {
    optional int64 requestId = 1;
    optional int64 userId = 2;
    optional string creationDate = 3;
}

message AuthenticateWebFailure {
    optional int64 requestId = 1;
    enum FailureReason {
        INVALID_CREDENTIALS = 0;
        INTERNAL_SERVER_ERROR = 1;
    }

    optional FailureReason reason = 2;
}